File: manuplationDigit/duckNo.py
number=int(input("enter your number :"))

lastdigit= number%10
thirdDigit=number//10%10
secondDigit= number//100%10
firstDigit= number//100

sum= firstDigit *1000 + secondDigit*100+ thirdDigit*10+ lastdigit*1
if secondDigit==0 or thirdDigit==0 or lastdigit==0:
    print("Duck number ")

elif firstDigit==0:
    print("Not a Duck Number")

else:
    print("Not a duck number ")
 

File: dsa/day1/duckno.cpp
#include<iostream>
using namespace std;
int main(){
int number=0,lastDigit,firstDigit,reverse,removesecondlast,secondlast,middleDigit,remove,sum;


    cout<<"entre the 4digit number";
    cin>>number;

    lastDigit=number%10;
    firstDigit=number/1000;
    remove=number/10;
    middleDigit=remove%10;
    removesecondlast=number/100;
    secondlast=removesecondlast%10;
   //sum=firstDigit+lastDigit+middleDigit;
    if(firstDigit!=0){
        cout<<"duck";
    }
    else if(lastDigit==0){
        cout<<"duck";
    }
    else if(middleDigit==0){
        cout<<"duck";
    } 
    else if(secondlast==0){
        cout<<"duck";
    }
    else{
        cout<<"not";
    }

    
    return 0;
}